mclagsyncd/mclaglink.h

```diff
@@ -90,36 +134,121 @@ struct mclag_fdb
 
 };
 
+
+//MCLAG Domain Key 
+struct mclagDomainEntry
+{
+   unsigned int domain_id;
+
+   mclagDomainEntry() {}
+   mclagDomainEntry(unsigned int id):domain_id(id) {}
+
+   bool operator <(const mclagDomainEntry &domain) const
+   {
+       return domain_id < domain.domain_id;
+   }
+
+   bool operator ==(const mclagDomainEntry &domain) const
+   {
+       return (domain_id == domain.domain_id);
+   }
+};
+
+
+//MCLAG Domain Data
+struct mclagDomainData
+{
+    std::string source_ip;
+    std::string peer_ip;
+    std::string peer_link;
+    int keepalive_interval;
+    int session_timeout;
+
+    mclagDomainData()
+    {
+        keepalive_interval = -1;
+        session_timeout    = -1;
+    }
+
+    bool mandatoryFieldsPresent() const
+    {
+        return (!source_ip.empty() && !peer_ip.empty());
+    }
+
+    bool allFieldsEmpty() const
+    {
+        return (source_ip.empty() && peer_ip.empty() && peer_link.empty() && keepalive_interval == -1 && session_timeout == -1);
+    }
+};
+
+typedef  std::tuple<std::string, std::string> vlan_mbr;
+
 class MclagLink : public Selectable {
 public:
     const int MSG_BATCH_SIZE;
+    std::map<std::string, std:: string> *p_learn;
     ProducerStateTable * p_port_tbl;
     ProducerStateTable * p_lag_tbl;
     ProducerStateTable * p_tnl_tbl;
     ProducerStateTable * p_intf_tbl;
     ProducerStateTable *p_fdb_tbl;
     ProducerStateTable *p_acl_table_tbl;
     ProducerStateTable *p_acl_rule_tbl;
+    Table *p_mclag_tbl;
+    Table *p_mclag_local_intf_tbl;
+    Table *p_mclag_remote_intf_tbl;
+    Table *p_device_metadata_tbl;
+    Table *p_mclag_cfg_table;
+    Table *p_mclag_intf_cfg_table;
+    Table *p_state_vlan_mbr_table;
+    Table *p_state_fdb_table;
     DBConnector *p_appl_db;
     DBConnector *p_asic_db; /*redis client access to ASIC_DB*/
     DBConnector *p_counters_db; /*redis client access to COUNTERS_DB*/
-    std::set <mclag_fdb> *p_old_fdb;
 
-    MclagLink(uint16_t port = MCLAG_DEFAULT_PORT);
+    ProducerStateTable *p_iso_grp_tbl;
+
+    SubscriberStateTable *p_mclag_intf_cfg_tbl;
+    SubscriberStateTable *p_state_fdb_tbl;
+    SubscriberStateTable *p_state_vlan_mbr_subscriber_table;
+
+    std::map<mclagDomainEntry, mclagDomainData> m_mclag_domains;
+
+    MclagLink(Select* select, int port = MCLAG_DEFAULT_PORT);
+
     virtual ~MclagLink();
 
     /* Wait for connection (blocking) */
     void accept();
 
     int getFd() override;
-    uint64_t readData() override;
+    char* getSendMsgBuffer();
+    int getConnSocket();
+    uint64_t readData() override; 
+    void mclagsyncd_fetch_system_mac_from_configdb();
```

Comment:
Is there a particular reason why the naming convention here is different from what was used before (lower camel case)? I believe it should be the same for all methods. Same for private methods below.

Reply:
addressed this review comment